lib/bunyan.js

```diff
@@ -221,61 +160,41 @@ function _indent(s, indent) {
  * @param dedupKey {String} Optional. A short string key for this warning to
  *      have its warning only printed once.
  */
+
+let _warned;
+
+if (!_warned) _warned = {};
+
 function _warn(msg, dedupKey) {
-    assert.ok(msg);
-    if (dedupKey) {
-        if (_warned[dedupKey]) {
-            return;
-        }
-        _warned[dedupKey] = true;
-    }
-    process.stderr.write(msg + '\n');
+  assert.ok(msg);
+  if (dedupKey) {
+    if (_warned[dedupKey]) return;
+    _warned[dedupKey] = true;
+  }
+  process.stderr.write(msg + '\n');
 }
 function _haveWarned(dedupKey) {
-    return _warned[dedupKey];
+  return _warned[dedupKey];
 }
-var _warned = {};
-
 
 function ConsoleRawStream() {}
-ConsoleRawStream.prototype.write = function (rec) {
-    if (rec.level < INFO) {
-        console.log(rec);
-    } else if (rec.level < WARN) {
-        console.info(rec);
-    } else if (rec.level < ERROR) {
-        console.warn(rec);
-    } else {
-        console.error(rec);
-    }
+ConsoleRawStream.prototype.write = (rec) => {
+  if (rec.level < INFO) {
+    console.log(rec);
+  } else if (rec.level < WARN) {
+    console.info(rec);
+  } else if (rec.level < ERROR) {
+    console.warn(rec);
+  } else {
+    console.error(rec);
+  }
 };
 
 
-//---- Levels
-
-var TRACE = 10;
-var DEBUG = 20;
-var INFO = 30;
-var WARN = 40;
-var ERROR = 50;
-var FATAL = 60;
-
-var levelFromName = {
-    'trace': TRACE,
-    'debug': DEBUG,
-    'info': INFO,
-    'warn': WARN,
-    'error': ERROR,
-    'fatal': FATAL
-};
-var nameFromLevel = {};
-Object.keys(levelFromName).forEach(function (name) {
-    nameFromLevel[levelFromName[name]] = name;
-});
-
 // Dtrace probes.
-var dtp = undefined;
-var probes = dtrace && {};
+// eslint-disable-next-line no-undef-init
```

Comment:
still needed even you have removed `= undefined`?

Reply:
Removed.